src/Microsoft.SqlTools.ServiceLayer/LanguageServices/LanguageService.cs

```diff
@@ -1212,6 +1212,28 @@ private bool RequiresReparse(ScriptParseInfo info, ScriptFile scriptFile)
                 || !string.Equals(prevSqlText, currentSqlText);
         }
 
+        private bool RefreshNeeded(string uri)
+        {
+            ConnectionInfo connInfo;
+            System.Data.Common.DbConnection connection;
+            connectionService.TryFindConnection(uri, out connInfo);
+            if (connInfo.ConnectionDetails.AuthenticationType != "AzureMFA") 
+            {
+                return false;
+            } 
+            else 
+            {
+                if (connInfo.TryGetConnection(ConnectionType.Default, out connection))
```

Comment:
How does this happen - shouldn't we have a connection already (after all the user had to connect the query in the first place)? Does it get cleared out if an error happens or something?

Reply:
https://github.com/microsoft/sqltoolsservice/blob/1b5ac4c017129b1f33dad10ce68b62f49d2d2800/src/Microsoft.SqlTools.ServiceLayer/LanguageServices/LanguageService.cs#L763 This line seems to indicate that I can use TryFindConnection but I'll do some more digging just to make sure

Reply:
Are you actually testing this with a connection that's timed out and needs to be refreshed?

Reply:
@Charles-Gagnon It seems like this will still return a connection even if the token has expired, so I'll need another way to check this.